pkg/operator/ceph/object/bucket/provisioner.go

```diff
@@ -157,7 +162,12 @@ func (p Provisioner) Grant(options *apibkt.BucketOptions) (*bktv1alpha1.ObjectBu
 		return nil, errors.Wrapf(err, "failed to get user %q", stats.Owner)
 	}
 
-	s3svc, err := object.NewS3Agent(objectUser.Keys[0].AccessKey, objectUser.Keys[0].SecretKey, p.getObjectStoreEndpoint(), logger.LevelAt(capnslog.DEBUG), p.tlsCert)
+	var s3svc *object.S3Agent
+	if p.insecureTLS {
+		s3svc, err = object.NewInsecureS3Agent(objectUser.Keys[0].AccessKey, objectUser.Keys[0].SecretKey, p.getObjectStoreEndpoint(), logger.LevelAt(capnslog.DEBUG))
```

Comment:
maybe add a debug mode log for both insecure and secure tls config?

Reply:
maybe no need to log this, since it's common behavior to use the different s3 agent.